perceval/backends/core/rss.py

```diff
@@ -0,0 +1,239 @@
+
+    This class retrieves the entries from a RSS feed.
+    To initialize this class the URL must be provided.
+    The `url` will be set as the origin of the data.
+
+    :param url: RSS url
+    :param tag: label used to mark the data
+    :param cache: cache object to store raw data
+    """
+    version = '0.1.0'
+
+    def __init__(self, url, tag=None, cache=None):
+        origin = url
+
+        super().__init__(origin, tag=tag, cache=cache)
+        self.url = url
+        self.client = RSSClient(url)
+
+    @metadata
+    def fetch(self):
+        """Fetch the entries from the url.
+
+        The method retrieves all entries from a RSS url
+
+        :returns: a generator of entries
+        """
+
+        logger.info("Looking for rss entries at feed '%s'", self.url)
+
+        self._purge_cache_queue()
+        nentries = 0  # number of entries
+
+        raw_entries = self.client.get_entries()
+        self._push_cache_queue(raw_entries)
+        self._flush_cache_queue()
+        entries = feedparser.parse(raw_entries)['entries']
+        for item in entries:
+            yield item
+            nentries += 1
+
+        logger.info("Total number of entries: %i", nentries)
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the entries from the cache.
+
+        :returns: a generator of entries
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_entries = next(self.cache.retrieve())
+        entries = feedparser.parse(cache_entries)['entries']
+
+        for item in entries:
+            yield item
+
+    @classmethod
+    def has_caching(cls):
+        """Returns whether it supports caching entries on the fetch process.
+
+        :returns: this backend supports entries cache
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend does not supports entries resuming
+        """
+        return False
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Build item."""
+        return str(item['link'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a RSS item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = str_to_datetime(item['published'])
+
+        return ts.timestamp()
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a RSS item.
+
+        This backend only generates one type of item which is
+        'item'.
+        """
+        return 'item'
```

Comment:
Is this the category for these items? In other places you talk about entries, not about items. Item is so generic, I think.